app/controllers/news_controller.rb

```diff
@@ -1,51 +1,39 @@
 class NewsController < ApplicationController
+  include Pagination
   before_action :set_news, only: [:show, :update, :destroy]
 
   # GET /news
   def index
-    @news = News.all
+    @pagy, @news = pagy(News.order(created_at: :desc))
 
-    render json: @news
+    render json: NewsSerializer.new(@news, pagination_options).serialized_json
   end
 
   # GET /news/1
   def show
-    render json: @news
+    render json: NewsSerializer.new(@news).serialized_json
   end
 
-  # POST /news
-  def create
-    @news = News.new(news_params)
+  # -ROUTE-> GET ?
+  #retrieving the news for the course of the logged user
+  def get_news_for_user
+    @course = get_auth_user.course
+    @pagy, @news = pagy(News.where(course: @course))
```

Comment:
Mi è venuto in mente che potremmo fare un query scope su user, che pesca tutte le news a lui indirizzate e le ordina per data. In questo modo ci torna easy pescarle con
```ruby
@pagy, @news = pagy(get_auth_user.news)
``` 
Dovrebbe essere fattibile
(Tra l'altro ```get_auth_user``` lo rinominerei in ```auth_user```)

Reply:
infatti aveva più senso anche per me

Reply:
auth_user non può essere un helper method che da già a disposizione rails?

la query corretta non è ```auth_user.news``` ma ```auth_user.course.news```